File: CC_P1/src/cc_p1/Leaf.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package cc_p1;

/**
 *
 * @author tim
 */
public class Leaf extends Node {
    
    private Token value;
    
    public Leaf(Token value) {
        super("");
        this.value = value;
    }

    /**
     * @return the value
     */
    public Token getValue() {
        return value;
    }

    /**
     * @param value the value to set
     */
    public void setValue(Token value) {
        this.value = value;
    }
    
    public void print(int level) {
        
        for(int i = 0; i < level; i++)
            System.out.print("\t");
        
        System.out.println(value.getType().toString() + " " + value.getValue().toString());
    }
    
}

File: CC_P2/Leaf.cpp

#include "Leaf.h"


Leaf::Leaf(std::string name, std::string value) : Node(name) {
 this->value = value;
}

std::string Leaf::getValue() const {
    return value;
}

void Leaf::setValue(std::string value) {
    this->value = value;
}

void Leaf::print(int level) const {
    for(int i = 0; i < level; i++)
            std::cout << '\t';

    std::cout << getName() << " " << value << std::endl;
}

